### DISCUSSION TITLE
Fail2ban would cut out the part with datetime matched the datepattern before it'd apply failregex

### DISCUSSION THREAD

**Discussion by colemar** [fail2ban/fail2ban#3912] [Q&A] [answered]
https://github.com/fail2ban/fail2ban/discussions/3137#discussioncomment-1532339

_Fail2ban would cut out the part with datetime matched the datepattern before it'd apply failregex_

Sorry, is this well documented?
Total surprise for me to discover the following:
```
# fail2ban-regex '2024-12-28T19:44:16+01:00|404|127.0.0.1|' '\|(40[03457]|50[01])\|<HOST>\|'
Failregex: 1 total

# fail2ban-regex '2024-12-28T19:44:16+01:00|404|127.0.0.1|' '.\|(40[03457]|50[01])\|<HOST>\|'
Failregex: 0 total

# fail2ban-regex '2024-12-28T19:44:16+01:00|404|127.0.0.1|' '^\|(40[03457]|50[01])\|<HOST>\|'
Failregex: 1 total
```

I admit that I didn't read the whole docs, but a web search did not show anything about this behaviour.

**Comment by sebres** [comment, marked as answer]
Yes it is documented and intended behavior - see [man jail.conf.5](https://manpages.debian.org/bookworm/fail2ban/jail.conf.5.en.html#failregex~2):

<i>NOTE: the failregex will be applied to the remaining part of message after prefregex processing (if specified), which in turn takes place after datepattern processing (<b>whereby the string of timestamp matching the best pattern, cut out from the message</b>).
</i>

The reason for that is: the format of timestamp (syslog or whatever) may deviate from system to system and may be distribution-, platform- or language-specific. Not to mention that messages read from systemd-journal would not have the timestamp at all (it is read a special field).
Also typical anchors in all stock filters for the messages starting with a timestamp looks like `^\s*...` (optional space after excised timestamp before the message text `...`).

> but a web search did not show anything about this behaviour

How so?
1. [site:github.com/fail2ban/fail2ban timestamp cut out](https://www.google.de/search?q=site%3Agithub.com%2Ffail2ban%2Ffail2ban+timestamp+cut+out)
2. [fail2ban timestamp "cut out"](https://www.google.de/search?q=fail2ban+timestamp+%22cut+out%22)
